### ISSUE TITLE
[Feature request] Option to add noise to blur

### ISSUE THREAD

**Issue by regern80** [yshui/compton#1015]
Noise adds beauty to blur and makes it stand out. Kwin does this in plasma, I think it'll be a nice feature in picom.  
Below is a screenshot blurred with noise by kwin
[Screenshot_20220322_140251](https://user-images.githubusercontent.com/81050351/215881590-4a223873-73e7-4175-b8b4-69747717cdfa.png)

**Comment by tryone144** [comment]
This has come up previously as a means to eliminate color-banding in https://github.com/yshui/picom/issues/602. Do I understand correctly that you would just like to have the noise for the aesthetic effect?

**Comment by absolutelynothelix** [comment]
@tryone144, yeah, it's a common practice. for example, windows also does this with their blur (i forgot how it's called), but this noise doesn't eliminate the color banding, so it's just an aesthetic effect:
![image](https://user-images.githubusercontent.com/52252283/215926073-51789e72-7220-4f0b-813a-5bfd8bd753a6.png)
so this feature makes sense and i think i'd like to see it implemented as well. maybe like this:
```
configuration file:
...
blur: {
    ...
    noise = true;
    ...
};
...

command line:
... --blur-noise ...
```
talking about an approach suggested by @yshui:
> we could add a custom present shader. by the looks of it it would be simple to do.

as far as i understand, currently dithering is implemented in the present shader what means that it's applied to the entire scene that will presented to a user. unfortunately, we can't apply noise in the present shader because in this case the entire scene will be noisy, we have to apply it on the blur texture (in the blur shader, maybe?).

p.s. however, a custom present shader was requested at least in the #684.

**Comment by CianJDuggan** [comment]
Having an option like window-shader-bg (background rather than foreground) to apply shaders at the blur level (behind text) could enable some nice subtle effects. Tinted glass (semi-transparent color layer), frosted glass (noise layer) and 3D accents (inner shadow/highlight borders) would be quite pretty! I'd love to take the glassmorphism in my UI to the next level. One could even run a simple python script in tandem with pywal to find the source of light in a wallpaper to create an immersive glass experience. Wishful thinking!
